### ISSUE TITLE
Handle keywords (NOT, AND, OR) in card search better

### ISSUE THREAD

**Issue by tooomm** [Cockatrice/Cockatrice#4039]
**Description of New Feature**
I had a chat with @ebbit1q today. We should think about removing the keyword `not` from card search.
There is support for `-` already to exclude/negate/inverse commands. The written keyword is, as reported by a user today, problematic with some cards like [Not of This World](https://scryfall.com/card/roe/8/not-of-this-world) for example.

**Additional Context**
It seems like we do not support the `and` keyword already.
So there is only one typical operator left: `or`.
Find a good solution for `or`... which single sign could express that? Maybe "|"? I'm not so sure about the mathematical "v" as it messes with the search.
Or only consider the keyword if written in all capital letters (`OR`)? That could apply for AND, and NOT too if needed.

**Comment by ebbit1q** [comment]
we don't need an and keyword, putting things next to each other is and.
I think making them capital should exclude them from the average search, most users don't know the word not does anything.

**Comment by ebbit1q** [comment]
Compare the search request for truth or tale on scryfall, it's not the only result because they too interpret or this way https://scryfall.com/search?q=truth%20or%20tale

**Comment by ZeldaZach** [comment]
cc: @basicer as this was your change and I'm curious what our best approach here is

**Comment by Pickle-boi** [comment]
I was discussing this with ebbit1q a week ago, and I think that the last fix proposed by tooomm is easily made. However, I am not a C++ programmer, so I might be completely off.

Looking at filter_string.cpp line 21, "NotQuery" is what is providing "not" as a valid argument in the search query. I am not very well versed in C++, but it appears that the parser is case sensitive, so changing this to "NOT" might fix it.

[https://github.com/Cockatrice/Cockatrice/blob/d4fc1be2ccaf7e76a91ce063aebe263a1a569183/cockatrice/src/filter_string.cpp#L21C7-L21C7](url)

**Comment by ebbit1q** [comment]
yes it would, we simply have to come to a consensus on what syntax we'd like

**Comment by tooomm** [comment]
Reading this again, it looks like we agree already that only considering keywords in all capital letters is a good solution.
--> `NOT` & `OR` 👍 

That should lead to:
Truth (and) or (and) Tale = Very limited result for this specific search
Truth OR Tale = Many results matching either search term

Scryfall very visibly helps the user to understand their results at the top of each search, that makes it a bit easier for users there:
`38 cards where the name includes “truth” or the name includes “tale”`
